cartographer/mapping/internal/2d/ray_casting.h

```diff
@@ -19,21 +19,16 @@
 
 #include <vector>
 
-#include "cartographer/common/port.h"
-#include "cartographer/mapping/2d/probability_grid.h"
-#include "cartographer/sensor/point_cloud.h"
-#include "cartographer/sensor/range_data.h"
 #include "cartographer/transform/transform.h"
 
 namespace cartographer {
 namespace mapping {
 
-// For each ray in 'range_data', inserts hits and misses into
-// 'probability_grid'. Hits are handled before misses.
-void CastRays(const sensor::RangeData& range_data,
-              const std::vector<uint16>& hit_table,
-              const std::vector<uint16>& miss_table, bool insert_free_space,
-              ProbabilityGrid* probability_grid);
+// Compute all pixels in which some part of the line segment connecting 'begin'
+// and 'end' lies with 'subpixel_scale' precision.
+std::vector<Eigen::Array2i> RayToPixelMask(const Eigen::Array2i& begin,
```

Comment:
rename to LineToPixelMask/Set or just RenderLine?
(after discussion has concluded, rename this file and the include guards accordingly)